modules/git/blame.go

```diff
@@ -98,17 +105,32 @@ func (r *BlameReader) Close() error {
 	r.bufferedReader = nil
 	_ = r.reader.Close()
 	_ = r.output.Close()
+	if r.ignoreRevsFile != nil {
+		_ = util.Remove(r.ignoreRevsFile.Name())
+	}
 	return err
 }
 
 // CreateBlameReader creates reader for given repository, commit and file
-func CreateBlameReader(ctx context.Context, repoPath, commitID, file string) (*BlameReader, error) {
-	cmd := NewCommandContextNoGlobals(ctx, "blame", "--porcelain").
-		AddDynamicArguments(commitID).
+func CreateBlameReader(ctx context.Context, repoPath string, commit *Commit, file string, bypassBlameIgnore bool) (*BlameReader, error) {
+	cmd := NewCommandContextNoGlobals(ctx, "blame", "--porcelain")
+
+	var ignoreRevsFile *os.File
+	if !bypassBlameIgnore {
+		ignoreRevsFile = tryCreateBlameIgnoreRevsFile(commit)
+		if ignoreRevsFile != nil {
+			cmd.AddOptionValues("--ignore-revs-file", ignoreRevsFile.Name())
+		}
```

Comment:
It would be better to use `--ignore-revs-file /dev/stdin` and pass the content of the `.git-blame-ignore-revs` file but there is no Windows equivalent for `/dev/stdin`.

Reply:
We could think about creating a unix and windows variant where the windows variant uses an actual file while the UNIX variant uses stdin…